aim/cli/reporting/reporter.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict
+import uuid
+
+from humbug.consent import HumbugConsent
+from humbug.report import Reporter
+
+from aim.__version__ import __version__ as aim_version
+from aim.engine.configs import HUMBUG_TOKEN, HUMBUG_KB_ID
+
+aim_version_tag = "version:{}".format(aim_version)
+aim_tags = [aim_version_tag]
+
+
+def configure_reporter(client_id: uuid.UUID, aim_consent: bool) -> Reporter:
+    """
+    Prepare Humbug Consent mechanism and create reporter.
+    """
+    session_id = str(uuid.uuid4())
+
+    aim_consent = HumbugConsent(aim_consent)
+    aim_reporter = Reporter(
```

Comment:
@kompotkot : You are creating asynchronous reporters, each of which creates a separate thread for reporting and registers its own `atexit` handler. So you should not do this inside a function.

It is better to create a reporter object in this module and import the object directly. Otherwise, every `configure_reporter` call will result in creation of separate thread and separate `Reporter` instance which will be stored in memory until program exits. This is bad if you are only going to use that reporter object a few times.

Reply:
If you want to use this functional style, set argument `mode = Modes.SYNCHRONOUS` when you instantiate reporter.